### DISCUSSION TITLE
Regarding Build-Time Plugin/Module Loading in NowInAndroid App

### DISCUSSION THREAD

**Discussion by GeorgePapageorgakis** [android/nowinandroid#1056] [General]
Hello,

I understand that NowInAndroid is a modularized application with distinct features, leading to the creation of two different apps based on the project's modules. My question is: Are these features integrated into the app's APK during the build process, or are they loaded dynamically during installation and execution?

Thank you.

**Comment by SimonMarquis** [comment]
Hi @GeorgePapageorgakis, this application currently does not use [Play Feature Delivery](https://developer.android.com/guide/playcore/feature-delivery) capabilities, and therefore all features are provided at install time.

**Reply by GeorgePapageorgakis** [reply]
So the built apk contains all features, (in a bundle like structure?), and during installation it decides what features to enable. Correct?

**Reply by SimonMarquis** [reply]
> So the built apk contains all features, (in a bundle like structure?)

Yes.

> during installation it decides what features to enable.

No, the whole package is installed.

**Reply by GeorgePapageorgakis** [reply]
So if understand correctly if I want to disable a feature, I would have to do so in the gradle build-logic and then the produced apk will not contain the feature right?? 

In short, we decide at build time, and build-logic module, what features we will include in the produced apk and then the produced apk will contain exactly the features that we included during the build time, right?

Thank you in advance.

**Reply by SimonMarquis** [reply]
This is not as easy as that. Please read the official documentation on [Play Feature Delivery](https://developer.android.com/guide/playcore/feature-delivery) first.

> In short, we decide at build time, and build-logic module, what features we will include in the produced apk and then the produced apk will contain exactly the features that we included during the build time, right?

At the moment, it's the `:app` module that decides what it includes in the final APK/AAB.

https://github.com/android/nowinandroid/blob/b3f5548fe2ad5d6bd926874f0f9f05a2f0d7dcfc/app/build.gradle.kts#L86-L91
